### DISCUSSION TITLE
Updating Node.js in the Zigbee2MQTT lxc

### DISCUSSION THREAD

**Discussion by KoalaWerewolf** [tteck/Proxmox#3658] [Q&A]
Hi,

I've recently had the situation where the update of z2m starts to complain about the version of Node.js. I tried all (that I know of) ways of updating the lxc to get it to automagically update the version of Node.js, but no luck. After some searching I found the /etc/apt/sources.list.d/nodesource.list config file where the repo for Node.js is specified, updating the URL allowed me to update Node.js.

Would it be possible to improve this amazing resource (ttecks scripts) to include an update of that source so Node.js automatically gets updated as needed? Or perhaps I've just missed something?

**Comment by Pocky0013** [comment]
Same thing for me. working with an LXC on Proxmox. I have updated Proxmox as well as Z2M with the "update" function, but I cannot get rid of this type of error:
`npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'minimatch@10.0.1',
npm warn EBADENGINE   required: { node: '20 || >=22' },
npm warn EBADENGINE   current: { node: 'v18.19.0', npm: '10.8.3' }`
